### PR TITLE
Issue #155552 task:Create a resource in tjreports API plugin to get the fields compatible with google DS for a report

### PR DESCRIPTION
**Pull request by vaivk369**
…he fields compatible with google DS for a report

### CODE REVIEW

File: src/reports/gsfields.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	/**
+	 * Function get filters data
+	 *
+	 * @return boolean
+	 */
+	public function get()
+	{
+		$app        = JFactory::getApplication();
+		$jinput     = $app->input;
+		$formData   = $jinput->post;
```
Comment (coolbung) [inline src/reports/gsfields.php]: Are you trying to use POST data in the get function ?

File: src/reports/gdsfields.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+{
+	/**
+	 * Function get fields
+	 *
+	 * @return boolean
+	 */
+	public function get()
+	{
+		$app        = JFactory::getApplication();
+		$jinput     = $app->input;
```
Comment (coolbung) [inline src/reports/gdsfields.php:32]: Is this variable used ?

File: src/reports/gdsfields.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+		if (!class_exists($className))
+		{
+			ApiError::raiseError(400, JText::_('PLG_API_REPORTS_REPORT_NAME_INVALID'), 'APIValidationException');
+		}
+
+		$reportPlugin = new $className;
+
+		// If plugin does not have getGoogleDsFields throw error
+		if (!method_exists($reportPlugin, 'getGoogleDsFields'))
```
Comment (coolbung) [inline src/reports/gdsfields.php]: Do you want to change the method name to `getGDSFields` ?

File: src/reports/gdsfields.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
+﻿<?php
+/**
+ * @package      Tjreports
+ * @subpackage   com_api
```
Comment (thite-amol) [inline src/reports/gdsfields.php]: @twsvaishali Change subpackage `com_api` to `API`

File: src/reports/gdsfields.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * @author       Techjoomla <extensions@techjoomla.com>
+ * @copyright    Copyright (C) 2009 - 2018 Techjoomla. All rights reserved.
```
Comment (thite-amol) [inline src/reports/gdsfields.php]: @twsvaishali Change the copyright year

File: src/reports/gdsfields.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+// No direct access.
+defined('_JEXEC') or die;
+
+/**
+ * Tjreports API Gs fields class
+ * This is used to get the fields for the Google studio connector
+ *
+ * @since  1.1.0
```
Comment (thite-amol) [inline src/reports/gdsfields.php]: @twsvaishali Use constant

File: src/reports/gdsfields.php [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+class ReportsApiResourceGdsfields extends ApiResource
+{
+	/**
+	 * Function get fields
+	 *
+	 * @return boolean
+	 */
+	public function get()
+	{
+		$app        = JFactory::getApplication();
```
Comment (thite-amol) [inline src/reports/gdsfields.php]: @twsvaishali Use the namespaces for all deprecated classes

File: src/reports/gdsfields.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+		{
+			ApiError::raiseError(400, JText::_('PLG_API_REPORTS_REPORT_NO_GOOGLESTUDIO_SUPPORT'), 'APIValidationException');
+		}
+
+		// Load language files
+		$lang = JFactory::getLanguage();
+		$lang->load('com_tjreports', JPATH_ADMINISTRATOR, 'en-GB', true);
+		$lang->load('plg_tjreports_' . $reportName, JPATH_SITE . "/plugins/tjreports/" . $reportName, 'en-GB', true);
+
+		$fields = $reportPlugin->getGDSFields();
```
Comment (thite-amol) [inline src/reports/gdsfields.php]: @twsvaishali Avoid this useless variable

File: src/reports/gdsfields.php [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+ * Tjreports API Gs fields class
+ * This is used to get the fields for the Google studio connector
+ *
+ * @since  1.1.0
+ */
+class ReportsApiResourceGdsfields extends ApiResource
+{
+	/**
+	 * Function get fields
+	 *
```
Comment (thite-amol) [inline src/reports/gdsfields.php:26]: @twsvaishali Update the function comments

**Review by thite-amol — CHANGES_REQUESTED** [review]

File: src/reports/report.php [hunk 10]
```diff
@@ -95,6 +97,7 @@ public function post()
 			}
 		}
 
+		$this->plugin->customAttributes->set("total", $reportPlugin->getTotal());
```
Comment (coolbung) [inline src/reports/report.php:100]: Don't think this is a good idea. We can't expect developers to know this internal attribute named `customAttributes`

Needs to be `$this->plugin->setCustomAttribute($key, $value)`

**Review by coolbung — CHANGES_REQUESTED** [review]
The custom attribute method needs to be updated

**Pull request merged**